javascript/require-and-import-mini-lesson.md

```diff
@@ -0,0 +1,351 @@
+- In a technical interview, you will be confident to answer the question: `When should you use Import instead of Require?`
+
+### Objectives
+
+**Participants will be able to:**
+
+- Differentiate between `require` and `import`
+- Be initiated to modular programming
+
+### Specific Things To Learn
+
+- [What are JavaScript modules?](#what-are-javascript-modules)
+- [What is require?](#what-is-require)
+- [What is import?](#what-is-import)
+- [Difference between import and require](#difference-between-import-and-require)
+- [Can we use import in node?](#can-we-use-import-in-node)
+- [Which one to use? And When?](#which-one-to-use?-and-when?)
+- [Chart of the Differences](#chart-of-the-differences)
+
+
+### Lesson
+
+With JavaScript, when you want to include code that another developer has produced and made available to the DEV community, you now two main ways to bring in that code into your own script: `require` and `import`. When starting out, it can sometimes become confusing when to use one or another. So, in order to clarify this a little bit, let's dive into the core of what they are, how they differ, and when you should use one or the other.
+
+#### What are JavaScript modules?
+
+The first thing we need to do is to understand what a JavaScript module is. Basically, it's a piece of code which is *exporting* objects that you can import in your own code. So, yes, `require` imports, and `import`, well, you've guessed it alright. The only thing is that they do it differently.
+
+If you want more information on modules, I invite you to have a look at the official MDN documentation in the [supplemental materials section](#supplemental-materials).
+
+#### What is require?
+
+In order to deal with JavaScript outside of the browser, Node.js follows the CommonJS module specification in which modules are loaded synchronously, i.e. one after the other. This is why you cannot use `require` as such in the browser, unless it is transpiled. A code is said to be transpiled when it is converted from a certain source code to another source code but producing the equivalent functionalities.
+
+In that perspective, when using Node, if you want to include modules that are in another file, you can use the `require` built-in function. It will first read a JavaScript file, then execute it, and finally return the `exports` object. Ultimately, the `require` function will look for the specified file in the following order:
+
+1. Built-in core Node.js modules. For example: `fs`
+
+2. Installed NPM Modules. It will look in the node_modules folder. For example: `express`
+
+3. Local Modules. 
+
+It is important to note that if the specified module name has a `./`, a `/` or a `../`, it will look for the directory/file in the given path. Please note that it will match the `.js`, `.json`, `.mjs`, `.cjs`, `.wasm` and `*.node` extensions.
```

Comment:
I would remove .mjs, .cjs, .wasm, and node extension since they are rarely seen.

It's more important to make the point that to require a file named `code.js` you just say `code` with no extension

Reply:
done in [c89273b](https://github.com/Techtonica/curriculum/pull/1414/commits/c89273b844a10c1f67a1d0886a773f18a1c327e8)